### PR TITLE
fix `poetry shell` for zsh if space character in venv path

### PR DESCRIPTION
**Pull request by dimstunt**
fix zsh: poetry shell doesn't work if space character in .venv path

# Pull Request Check List

Resolves: #7244

Confirmed working on bash, zsh

### CODE REVIEW

**Review by PabloEmidio — DISMISSED** [review]

**Review by PabloEmidio — COMMENTED** [review]
On [PEP 8](https://peps.python.org/pep-0008/#string-quotes), there is the following say about triple quote

> For triple-quoted strings, always use double quote characters to be consistent with the docstring convention in [PEP 257](https://peps.python.org/pep-0257).

**Comment by jaklan** [comment]
Or even simpler:
`c.sendline(f'emulate bash -c ". {shlex.quote(str(activate_path))}"')`
(so we get `emulate bash -c ". '/some path/activate'"`)

But the safest one would be:
`c.sendline(f"emulate bash -c {shlex.quote(f'. {shlex.quote(str(activate_path))}')}")`
(so we get `emulate bash -c '. '"'"'/some path/activate'"'"''`)

**Comment by DennisTheMenace780** [comment]
Any updates on whether this will get merged in the near future? I am running into the same issue on macOS unfortuantely.

**Comment by dimstunt** [comment]
@PabloEmidio when you have a chance, please check this.

**Comment by PabloEmidio** [comment]
> @PabloEmidio when you have a chance, please check this.

For what I saw, it should do it. Could any test be added to validate this scenary?

**Comment by dimstunt** [comment]
> > @PabloEmidio when you have a chance, please check this.
> 
> For what I saw, it should do it. Could any test be added to validate this scenary?

Not fast because the executing bash script is moked, and I don't have enough expertise in pytest to change that.

File: src/poetry/utils/shell.py [hunk 1]
```diff
@@ -122,7 +122,10 @@ def activate(self, env: VirtualEnv) -> int | None:
 
         if self._name == "zsh":
             # Under ZSH the source command should be invoked in zsh's bash emulator
-            c.sendline(f"emulate bash -c '. {shlex.quote(str(activate_path))}'")
+            quoted_activate_path = shlex.quote(str(activate_path))
+                c.sendline(
+                    f"emulate bash -c {shlex.quote(f'. {quoted_activate_path}')}"
+                )
```
Comment (radoering) [inline src/poetry/utils/shell.py:128]:
```suggestion
            quoted_activate_path = shlex.quote(str(activate_path))
            c.sendline(
                f"emulate bash -c {shlex.quote(f'. {quoted_activate_path}')}"
            )
```

fix merge mistake

**Comment by radoering** [comment]
pre-commit.ci autofix

**Review by radoering — APPROVED** [review]

**Pull request merged**